client/pages/appointmentTable.jsx

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import AppointmentForm from './appointmentForm';
+import Table from 'react-bootstrap/Table';
+
+export default class AppointmentTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userData: this.props.userData,
+      monthIndex: null,
+      selectedDate: null,
+      selectedDay: null,
+      firstDayIndex: 0,
+      firstDay: null,
+      lastDate: null,
+      openModal: false,
+      result: null,
+      table: null,
+      selectDayAttps: null
+    };
+    this.dayCallBack = this.dayCallBack.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.findDay = this.findDay.bind(this);
+    this.createTable = this.createTable.bind(this);
+  }
+
+  dayCallBack(event) {
+    const selectedDate = parseInt(event.target.textContent);
+    const selectedDay = this.findDay(selectedDate);
+
+    const selectDayAttps = this.state.result[selectedDate - 1].time;
+
+    fetch('/api/appointment')
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      });
+
+    this.setState({ selectedDate, selectedDay, openModal: true, selectDayAttps });
+  }
+
+  findDay(date) {
+    const weekday = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```

Comment:
Since you use this it multiple places you can add it as a property on your class in the constructor.